frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -206,7 +210,27 @@ export const NavBar = withRouter<NavBarProps>(({ pages, baseURL, basePath }) =>
     </>
   );
 
-  return <ul className="co-m-horizontal-nav__menu">{tabs}</ul>;
+  const activePage = pages.find(({ href, path }) => {
+    const matchURL = matchPath(location.pathname, {
+      path: `${basePath}/${removeLeadingSlash(path || href)}`,
+      exact: true,
+    });
+    return matchURL?.isExact;
+  });
+
+  const labelId = activePage?.nameKey?.split('~')[1] || activePage?.name || 'Details';
+  return (
+    <>
+      <Helmet>
+        <title data-telemetry={telemetryPrefix ? `${telemetryPrefix} · ${labelId}` : labelId}>
+          {titlePrefix
+            ? `${titlePrefix} · ${t(activePage?.nameKey) || activePage?.name}`
+            : `${t(activePage?.nameKey) || activePage?.name}`}
```

Comment:
Normally we don't call t() when nameKey is null or undefined. But it's working fine and doesn't log a warning/error, so it's okayish for me. But I would prefer:

```suggestion
            ? `${titlePrefix} · ${activePage?.nameKey ? t(activePage.nameKey) : activePage?.name}`
            : `${activePage?.nameKey ? t(activePage.nameKey) : activePage?.name}`}
```